lesson_14/exceptions/exceptions_app/src/main/java/com/codedifferently/lesson14/ecommerce/EcommerceSystem.java

```diff
@@ -28,8 +40,20 @@ public void cancelOrder(String orderId) {
     orders.remove(orderId);
   }
 
-  public String checkOrderStatus(String orderId) {
+  public String checkOrderStatus(String orderId)
+      throws OrderNotFoundException, ProductNotFoundException {
+
     Order order = orders.get(orderId);
+    // Since order is set at this exact moment all we need to do is check if the given id is in the
```

Comment:
Use proper casing, grammar, and punctuation for your comments. Also, use shorter sentences for clarity. More importantly, I'd suggest that the code is clear enough on its own, so adding it just makes your reader have to do more work.

In general, only add comments to explain something that isn't immediately obvious in the code and that can't be made any simpler by rewriting the code itself.